src/main/java/frc/bionic/swerve/Vision.java

```diff
@@ -0,0 +1,262 @@
+  private double tl; 
+
+  // Front light object
+  private double frontLight;
+  // Back light object 
+  private double backLight;
+  // The value the API uses to differentiate between what the user sees
+  private double pipeline = 0;
+
+  public Vision(double slope, double yintercept, double targetHeight, double cameraHeight, double cameraAngle) {
+    super();
+    this.slope = slope;
+    this.yintercept = yintercept;
+    this.targetHeight = targetHeight;
+    this.cameraHeight = cameraHeight;
+    this.cameraAngle = cameraAngle;
+
+    this.frontCamFeed = NetworkTableInstance.getDefault().getTable("limelight");
+    this.setFrontCamFeed();
+    this.updateTableVariables();
+  }  
+  
+  double ledState = 1;
+
+  @Override
+  public void periodic() {
+    frontCamFeed.getEntry("ledMode").setNumber(ledState);
+    frontCamFeed.getEntry("pipeline").setNumber(pipeline);
+  }
+
+  /**
+   * Updates network frontCamFeed variables.
+   */
+  private void updateTableVariables() {
+    tv = activeCamFeed.getEntry("tv").getDouble(0.0);
+    tx = activeCamFeed.getEntry("tx").getDouble(0.0);
+    ty = activeCamFeed.getEntry("ty").getDouble(0.0);
+    ta = activeCamFeed.getEntry("ta").getDouble(0.0);
+    ts = activeCamFeed.getEntry("ts").getDouble(0.0);
+    tl = activeCamFeed.getEntry("tl").getDouble(0.0);
+  }
+
+  /**
+   * Sets the led states
+  */
+  public void setLights(final double ledState){
+    this.ledState = ledState;
+  }
+
+  /**
+   * 
+   * Establishes a pipeline based on what the user sees
+  */
+  public void setPipeline(final double pipeline){
+    this.pipeline = pipeline;
+  }
+
+  /**
+   * Gets the X offset in degrees.
+   *
+   * @return The X offset in degrees.
+   */
+  public double getXOffset() {
+    return tx;
+  }
+
+  /**
+   * Gets the Y offset in radians.
+   *
+   * @return The Y offset in radians.
+   */
+  public double getYOffset() {
+    return ty;
+  }
+
+  /**
+   * Gets the target area.
+   *
+   * @return The target area.
+   */
+  public double getTargetArea() {
+    return ta;
+  }
+
+  /**
+   * Gets the target skew.
+   *
+   * @return Target Skew.
+   */
+  public double getTargetSkew() {
+    return ts;
+  }
+
+  /**
+   * Gets the X limelight's latency.
+   *
+   * @return Latency
+   */
+  public double getLatency() {
+    return tl;
+  }
+
+  /**
+   * Returns 1 if a target is visible, returns 0 if there's no target visible.
+   *
+   * @return 1 or 0
+   */
+  public double isTargetVisibleDouble() {
+    return tv;
+  }
+
+  /**
+   * Gets the distance of the target using the formula from the limelight's documentation.
+   *
+   * @return The distance to the target using a formula.
+   */
+  public double calculateDistanceFromCameraHeight(final double targetHeight, final double cameraHeight, final double cameraAngle) {
+    final double methodDistance = (targetHeight - cameraHeight) / Math.tan(Math.toRadians(cameraAngle + getYOffset()));
+    return methodDistance;
```

Comment:
`methodDistance`? What is the meaning of the term "method" in this context? Should this be `targetDistance` as per the method documentation?